### PR TITLE
Add tower-hyper for consul

### PR DESCRIPTION
**Pull request by LucioFranco**
Still a few warnings left to clean up but this works.

### CODE REVIEW

**Comment by LucioFranco** [comment]
@hntd187 it removes the old way of using hyper with the new one with the work that I have been doing in tower. But still needs a way to translate a stream into a concrete set of bytes which I have not found the right way to do. so its small progress in a refactor.

File: src/bin/toshi.rs [hunk 1]
```diff
@@ -192,22 +198,35 @@ fn run(catalog: Arc<RwLock<IndexCatalog>>, settings: &Settings) -> impl Future<I
...
-            .and_then(|_| cluster::init_node_id(settings_path))
-            .and_then(move |id| {
-                consul_client.set_node_id(id);
-                consul_client.register_node()
+            .and_then(|mut consul_client| {
+                let mut client_clone = consul_client.clone();
+                consul_client
+                    .register_cluster()
+                    .and_then(move |_| cluster::read_node_id(settings_path_read.as_str()))
+                    .then(|result| match result {
```
Comment (mauri870) [inline src/bin/toshi.rs]: This stuff is into cluster::init_node_id now. You can either use this function or remove it.

Comment (LucioFranco) [reply src/bin/toshi.rs]: 👍 Right, I saw that during the rebase but was all over the place so I think I overwrote that 😅 . I will revert to that change, thanks for pointing that out 👍

Comment (mauri870) [reply src/bin/toshi.rs]: Dunno if init_node_id is a good name for this... Maybe rename it to setup_node_id or something.

**Comment by hntd187** [comment]
I know I'm super late to the game here, but would this allow us to remote h2 generally and use hype for all the backend tower stuff?

**Comment by LucioFranco** [comment]
@hntd187 so pretty much yes, hypers http2 is all h2 so we could totally drop tower-h2 and use hyper for all http1 and http2 requests that was the goal. Linkerd also uses hyper for everything.

File: src/cluster/consul.rs [hunk 2]
```diff
@@ -147,49 +146,64 @@ impl Builder {
...
-
-        Ok(Consul {
-            address,
-            scheme,
-            client,
-            cluster_name: self.cluster_name.unwrap_or_else(|| "kitsune".into()),
-            node_id: self.node_id.unwrap_or_else(|| "alpha".into()),
+    pub fn build(self) -> impl Future<Item = Consul, Error = ClusterError> {
+        future::lazy(|| {
+            let uri = self.destination.unwrap_or_else(|| Uri::from_static("http://127.0.0.1:8500"));
```
Comment (mauri870) [inline src/cluster/consul.rs]: What about grabbing these defaults from `Settings::default_consul_addr()`, `Settings::default_cluster_name()` and so on?

Comment (LucioFranco) [reply src/cluster/consul.rs]: That is a good idea!

**Comment by LucioFranco** [comment]
@hntd187 did we want to continue with this or shall i close it?

**Comment by hntd187** [comment]
@LucioFranco I think we def want to use hyper for all the server impls here, but it seems like doing that with all the other tower components is a WIP right now, so it probably is better to be consistent right now and just stick with what we have now.

**Comment by LucioFranco** [comment]
@hntd187 ok I can close, though tower-hyper works with all the middleware the only issue is tower-grpc + tower-hyper.

**Pull request closed without merging**